x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/transport/TransportFindFieldStructureAction.java

```diff
@@ -0,0 +1,93 @@
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.ParentTaskAssigningClient;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.ml.utils.MapHelper;
+import org.elasticsearch.xpack.core.textstructure.action.AbstractFindStructureRequest;
+import org.elasticsearch.xpack.core.textstructure.action.FindFieldStructureAction;
+import org.elasticsearch.xpack.core.textstructure.action.FindStructureResponse;
+import org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinder;
+import org.elasticsearch.xpack.textstructure.structurefinder.TextStructureFinderManager;
+import org.elasticsearch.xpack.textstructure.structurefinder.TextStructureOverrides;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class TransportFindFieldStructureAction extends HandledTransportAction<FindFieldStructureAction.Request, FindStructureResponse> {
+
+    private final Client client;
+    private final TransportService transportService;
+    private final ThreadPool threadPool;
+
+    @Inject
+    public TransportFindFieldStructureAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        ThreadPool threadPool
+    ) {
+        super(FindFieldStructureAction.NAME, transportService, actionFilters, FindFieldStructureAction.Request::new, threadPool.generic());
+        this.client = client;
+        this.transportService = transportService;
+        this.threadPool = threadPool;
+    }
+
+    @Override
+    protected void doExecute(Task task, FindFieldStructureAction.Request request, ActionListener<FindStructureResponse> listener) {
+        TaskId taskId = new TaskId(transportService.getLocalNode().getId(), task.getId());
+        new ParentTaskAssigningClient(client, taskId).prepareSearch(request.getIndex())
+            .setSize(request.getLinesToSample())
+            .setFetchSource(true)
+            .setQuery(QueryBuilders.existsQuery(request.getField()))
+            .setFetchSource(new String[] { request.getField() }, null)
+            .execute(ActionListener.wrap(searchResponse -> {
+                long hitCount = searchResponse.getHits().getHits().length;
+                if (hitCount < AbstractFindStructureRequest.MIN_SAMPLE_LINE_COUNT) {
+                    listener.onFailure(
+                        new IllegalArgumentException("Input contained too few lines [" + hitCount + "] to obtain a meaningful sample")
+                    );
```

Comment:
```suggestion
                    );
                    return;
```

(Or put the code below in an `else` branch if you prefer.)